lib/spack/spack/fetch_strategy.py

```diff
@@ -1247,7 +1233,10 @@ def _from_merged_attrs(fetcher, pkg, version):
     else:
         url = getattr(pkg, fetcher.url_attr)
 
-    attrs = {fetcher.url_attr: url}
+    attrs = {
+        fetcher.url_attr: url,
+        'mirrors': getattr(pkg, 'mirrors', None)
+    }
```

Comment:
This needs to go through pattern substitution as `url` did - so there's at least a test and the implementation that make it pass missing.  Anyhow, I submitted a draft early with only one package changed to get some input how people would receive a feature like this. Any comment is welcome.

Reply:
Added in 8162844